prow/config/cache.go

```diff
@@ -20,14 +20,76 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/cache"
 	"k8s.io/test-infra/prow/git/v2"
 )
 
+// MkEvictCallback constructs a simplelru.EvictCallback function to be used with
+// an InRepoConfigCache. It requires the name of the component and the evictions
+// metric to be used for the callback.
+func MkEvictCallback(component string) simplelru.EvictCallback {
+	evictionsMetric := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: fmt.Sprintf("%s_inRepoConfigCache_evictions", component),
+		Help: "Count of cache evictions by org and repo.",
+	}, []string{
+		"component",
+		"org",
+		"repo",
+	})
+
+	prometheus.MustRegister(evictionsMetric)
```

Comment:
We should do this once in an `init` block like we do with other metrics both for consistency and to prevent this from accidentally being registered twice. Alternatively you could use `sync.Once`, but using init seems simpler and it isn't really problematic for this metric to be served from components that don't use inrepoconfig.